staging/src/k8s.io/apiserver/pkg/util/flowcontrol/fairqueuing/fairqueuing.go

```diff
@@ -59,7 +59,7 @@ func (qsf queueSetFactoryImpl) NewQueueSet(concurrencyLimit, desiredNumQueues, q
 // https://github.com/kubernetes/enhancements/blob/master/keps/sig-api-machinery/20190228-priority-and-fairness.md
 type queueSetImpl struct {
 	lock                 sync.Mutex
-	wg                   *sync.WaitGroup
+	wg                   OptionalWaitGroup
```

Comment:
plants prometheus metrics instead?

Reply:
Huh?  This does nothing with Prometheus.

Reply:
The reason for a queue set to have an optional wait group is to support testing like is done in https://github.com/kubernetes/kubernetes/blob/feature-rate-limiting/staging/src/k8s.io/apiserver/pkg/util/flowcontrol/fairqueuing/fq_test.go .  The test function needs to know when all the asynchronous work has finished before advancing the fake clock.

Reply:
oh misread..